privilege/privileges/privileges.go

```diff
@@ -166,6 +166,9 @@ func (p *UserPrivileges) Check(ctx context.Context, db *model.DBInfo, tbl *model
 		if len(p.User) == 0 {
 			// User current user
 			p.User = variable.GetSessionVars(ctx).User
+			if len(p.User) == 0 {
```

Comment:
For tidb-server mode, we force user do login and auth. In this situation, len(variable.GetSessionVars(ctx).User) > 0.
But in other cases (embeded-db, unittest) we do not do login and auth. So a lot of testcase will be failed. Maybe I will remove this check latter.
I will add some comments now.